pkg/net/libp2p/libp2p_transport.go

```diff
@@ -0,0 +1,500 @@
+	if !found {
+		t.logger.Log(logging.LevelWarn, "addOutboundStream: failed to find the node", "src", t.ownID, "dst", nodeID)
+		return
+	}
+	node.Stream = s
+}
+
+func (t *Transport) streamExists(nodeID types.NodeID) bool {
+	t.nodesLock.RLock()
+	defer t.nodesLock.RUnlock()
+
+	v, found := t.nodes[nodeID]
+	return found && v.Stream != nil
+}
+
+func (t *Transport) getNodeStream(nodeID types.NodeID) (network.Stream, error) {
+	t.nodesLock.RLock()
+	defer t.nodesLock.RUnlock()
+
+	_, found := t.nodes[nodeID]
+	if !found {
+		return nil, &nodeInfoError{nodeID}
+	}
+
+	node, found := t.nodes[nodeID]
+	if !found ||
+		node.Stream == nil {
+		return nil, &nilStreamError{SrcNode: t.ownID, DestNode: nodeID}
+	}
+	return node.Stream, nil
+}
+
+func (t *Transport) getNodeAddr(nodeID types.NodeID) (types.NodeAddress, bool) {
+	t.nodesLock.RLock()
+	defer t.nodesLock.RUnlock()
+
+	node, found := t.nodes[nodeID]
+	if !found {
+		return nil, false
+	}
+	return node.Addr, true
+}
+
+func (t *Transport) isConnectionToNodeInProgress(nodeID types.NodeID) bool {
+	t.nodesLock.RLock()
+	defer t.nodesLock.RUnlock()
+
+	info, found := t.nodes[nodeID]
+	return found && info.IsOpening
+}
+
+func (t *Transport) setConnectionInProgress(nodeID types.NodeID) {
+	t.nodesLock.Lock()
+	defer t.nodesLock.Unlock()
+
+	info, found := t.nodes[nodeID]
+	if !found {
+		return
+	}
+	info.IsOpening = true
+}
+
+func (t *Transport) clearConnectionInProgress(nodeID types.NodeID) {
+	t.nodesLock.Lock()
+	defer t.nodesLock.Unlock()
+
+	info, found := t.nodes[nodeID]
+	if !found {
+		return
+	}
+	info.IsOpening = false
+}
+
+func (t *Transport) ImplementsModule() {}
+
+func (t *Transport) EventsOut() <-chan *events.EventList {
+	return t.incomingMessages
+}
+
+func (t *Transport) ApplyEvents(ctx context.Context, eventList *events.EventList) error {
+	iter := eventList.Iterator()
+	for event := iter.Next(); event != nil; event = iter.Next() {
+
+		switch e := event.Type.(type) {
+		case *eventpb.Event_Init:
+			// no actions on init
+		case *eventpb.Event_SendMessage:
+			for _, destID := range e.SendMessage.Destinations {
+				if types.NodeID(destID) == t.ownID {
+					// Send message to myself bypassing the network.
+					// The sending must be done in its own goroutine in case writing to gt.incomingMessages blocks.
+					// (Processing of input events must be non-blocking.)
+					receivedEvent := events.MessageReceived(
+						types.ModuleID(e.SendMessage.Msg.DestModule),
+						t.ownID,
+						e.SendMessage.Msg,
+					)
+					go func() {
+						select {
+						case t.incomingMessages <- events.ListOf(receivedEvent):
+						case <-ctx.Done():
+						}
+					}()
+				} else {
+					// Send message to another node.
+					if err := t.Send(ctx, types.NodeID(destID), e.SendMessage.Msg); err != nil { // nolint
+						// TODO: Handle sending errors (and remove "nolint" comment above).
+						//       Also, this violates the non-blocking operation of ApplyEvents method. Fix it.
```

Comment:
I don't get this. Isn't this a "fire-and-forget"? We don't wait for a response, do we?

Reply:
Fixed. It was legacy code